src/main/java/io/anserini/qa/RetrieveSentences.java

```diff
@@ -0,0 +1,213 @@
+
+  public RetrieveSentences(QAargs qAargs) throws Exception {
+    Path indexPath = Paths.get(qAargs.index);
+
+    if (!Files.exists(indexPath) || !Files.isDirectory(indexPath) || !Files.isReadable(indexPath)) {
+      throw new IllegalArgumentException(qAargs.index + " does not exist or is not a directory.");
+    }
+
+    this.reader = DirectoryReader.open(FSDirectory.open(indexPath));
+    this.passageClass = Class.forName("io.anserini.qa.passage." + qAargs.scorer + "PassageScorer");
+    scorer = (PassageScorer) this.passageClass.newInstance();
+  }
+
+  public void search(SortedMap<Integer, String> topics, String submissionFile, int numHits)
+          throws IOException, ParseException {
+    IndexSearcher searcher = new IndexSearcher(reader);
+
+    //using BM25 scoring model
+    Similarity similarity = new BM25Similarity(0.9f, 0.4f);
+    searcher.setSimilarity(similarity);
+
+    PrintWriter out = new PrintWriter(Files.newBufferedWriter(Paths.get(submissionFile), StandardCharsets.US_ASCII));
+
+    EnglishAnalyzer ea = new EnglishAnalyzer();
+    QueryParser queryParser = new QueryParser(FIELD_BODY, ea);
+    queryParser.setDefaultOperator(QueryParser.Operator.OR);
+
+    for (Map.Entry<Integer, String> entry : topics.entrySet()) {
+      int qID = entry.getKey();
+      String queryString = entry.getValue();
+      Query query = AnalyzerUtils.buildBagOfWordsQuery(FIELD_BODY, ea, queryString);
+
+      TopDocs rs = searcher.search(query, numHits);
+      ScoreDoc[] hits = rs.scoreDocs;
+      ScoredDocuments docs = ScoredDocuments.fromTopDocs(rs, searcher);
+
+      for (int i = 0; i < docs.documents.length; i++) {
+        out.println(String.format("%d %s %d %f", qID,
+                docs.documents[i].getField(FIELD_ID).stringValue(), (i + 1), docs.scores[i]));
+      }
+    }
+    out.flush();
+    out.close();
+  }
+
+  public void getRankedPassages(QAargs qaArgs) throws Exception {
+    IndexUtils util = new IndexUtils(qaArgs.index);
+    List<String> sentencesList = new ArrayList<>();
+    try (BufferedReader br = new BufferedReader(new FileReader(qaArgs.output))) {
+      String line;
+
+      while ((line = br.readLine()) != null) {
+        String docid = line.trim().split(" ")[1];
+        List<Sentence> sentences = util.getSentDocument(docid);
+        for (Sentence sent : sentences) {
+          sentencesList.add(sent.text());
+        }
+      }
+    }
+    Context c = new Context();
+    scorer.score(sentencesList, qaArgs.index, qaArgs.output, c);
+    Map<String, Double> sentences = c.getState();
+
+    Map.Entry<String, Double> maxEntry = null;
+    for (Map.Entry<String, Double> entry : sentences.entrySet()) {
+      if (maxEntry == null || entry.getValue().compareTo(maxEntry.getValue()) > 0) {
+        maxEntry = entry;
+      }
+    }
+    System.out.println("Answer: " + maxEntry.getKey());
+  }
+
+  public SortedMap<Integer, String> readTopicsFile(String topicsFile) throws IOException {
```

Comment:
Refactor this out into a class, put inside `io.anserini.search.query`. Follow conventions there.